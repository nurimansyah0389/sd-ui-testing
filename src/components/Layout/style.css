.Main {
    width: 840px;
    margin: 0 auto;
}
.Main__Logo {
    margin-top: 50px;
    margin-bottom: 50px;
}
.Main__Logo > img {
    width: 360px;
    margin: 0 auto;
    display: block;
}
.Main__Header > h1 {
    text-align: center;
    color: var(--color-primary);
    border-bottom: 1px solid var(--color-primary);
    padding-bottom: 20px;
}
.Main__Container {
    margin-bottom: 50px;
}