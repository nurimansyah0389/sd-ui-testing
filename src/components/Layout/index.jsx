import React from 'react';
import PropTypes from 'prop-types';

import './style.css';

const Layout = ({ children }) => <main className="Main">
    <div className="Main__Logo">
        <img src="/assets/images/logo.png" alt=""/>
    </div>
    <div className="Main__Header">
        <h1>Snapshot Testing Showcase</h1>
    </div>
    <div className="Main__Container">{children}</div>
</main>;
Layout.propTypes = {
    children: PropTypes.node
};
export default Layout;