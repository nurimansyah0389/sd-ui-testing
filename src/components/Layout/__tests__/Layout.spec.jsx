import React from 'react';
import Renderer from 'react-test-renderer';

import Layout from '../';

describe('Layout Spec', () => {
    it('Should render children without error', () => {
        // Assign
        const tree = Renderer.create(<Layout>Hello World</Layout>);

        // Assert
        expect(tree.toJSON()).toMatchSnapshot();
    });
});