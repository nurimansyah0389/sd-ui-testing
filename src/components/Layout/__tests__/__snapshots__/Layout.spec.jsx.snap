// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Spec Should render children without error 1`] = `
<main
  className="Main"
>
  <div
    className="Main__Logo"
  >
    <img
      alt=""
      src="/assets/images/logo.png"
    />
  </div>
  <div
    className="Main__Header"
  >
    <h1>
      Snapshot Testing Showcase
    </h1>
  </div>
  <div
    className="Main__Container"
  >
    Hello World
  </div>
</main>
`;
